client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -19,9 +19,25 @@ import EditorMediaModalDetailPreviewAudio from './detail-preview-audio';
 import EditorMediaModalDetailPreviewDocument from './detail-preview-document';
 import Button from 'components/button';
 import { userCan, isJetpack } from 'lib/site/utils';
+import versionCompare from 'lib/version-compare';
 import MediaUtils, { isItemBeingUploaded } from 'lib/media/utils';
 import config from 'config';
 
+/**
+ * This function returns true if the site is a Jetpack site
+ * and its version is greater than the minimum version
+ * to be able to edit media files.
+ *
+ * @param  {object} site - current site
+ * @return {boolean} `true` is it's a valid Jetpack site
+ */
+const allowForJetpackSite = site => (
+	site &&
+	isJetpack( site ) &&
+	site.options &&
```

Comment:
Maybe some lodash get usage here can reduce the number of undefined checks we need to make.
```jsx
const isJetpack = get( site, 'jetpack', false );
if ( ! isJetpack ) {
    return false;
}
const jetpackVersion = get( site, 'options.jetpack_version', '0.0' );
return versionCompare( jetpackVersion, '4.7-alpha', '>=' );
```